### PR TITLE
distribute LP rewards

### PR DESCRIPTION
**Pull request by samparsky**

### CODE REVIEW

File: scripts/distribute-lp-rewards/cli.js [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+const coreAddr = cfg.ETHEREUM_CORE_ADDR
+
+const DISTRIBUTION_STARTS = new Date('2020-08-04T00:00:00.000Z')
+const DISTRIBUTION_ENDS = new Date('2020-12-31T00:00:00.000Z')
+
+const INCENTIVE_CHANNEL_OPEN_FEE = bigNumberify('1500000000000000000')
+const INCENTIVE_TO_DISTRIBUTE = bigNumberify('7010000000000000000000000')
+const DISTRIBUTION_PER_SECOND = bigNumberify('478927203065134100')
+
```
Comment (IvoPaunov) [inline scripts/distribute-lp-rewards/cli.js:37]: Just to pin double check for distribution constants - start, end, amount, per second, etc.

File: scripts/distribute-lp-rewards/lib.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+function getDistributionForPeriodWithMultiplier(
+	distribution = {},
+	parsedLogs,
+	startSeconds,
+	endSeconds,
+	distributionPerSecond,
+	parseEventLog,
+	liquiditySupplyDurationByMultiplier
+) {
+	const ONE_WEEK = 604800
```
Comment (IvoPaunov) [inline scripts/distribute-lp-rewards/lib.js:35]: can be exported as lib `const` (and used in the tests)

File: test/distribute-lp-reward.js [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+	// 2 weeks between DISTRIBUTION_START & DISTRIBUTION_ENDS
+	const DISTRIBUTION_START = new Date()
+	DISTRIBUTION_START.setDate(DISTRIBUTION_START.getDate() - 15)
+
+	const DISTRIBUTION_ENDS = new Date()
+	DISTRIBUTION_ENDS.setDate(DISTRIBUTION_ENDS.getDate() - 1)
+
+	const addDaysToDistributionStart = addDays(DISTRIBUTION_START)
+
+	const testBalancerTransferEvents = [
```
Comment (IvoPaunov) [inline test/distribute-lp-reward.js:130]: - `testBalancerTransferEvents` and `testUniswapTransferEvents` are absolutely the same.
can be declared as one var. 
- maybe one more test is needed - to check the distribution for `uniswap` and `balancer` at the same time as it is in the real usecase

File: test/distribute-lp-reward.js [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+	]
+
+	const liquidityDurationByMultiplier = {
+		1: 10, // 1 week => 0.1% increase
+		2: 20,
+		3: 30
+	}
+
+	const distributionStarts = Math.floor(DISTRIBUTION_START.getTime() / 1000)
+	const distributionEnds = DISTRIBUTION_ENDS.getTime() / 1000
```
Comment (IvoPaunov) [inline test/distribute-lp-reward.js:89]: - `Math.floor`
- distribution params can be common for all the test as they are the same

**Review by IvoPaunov — COMMENTED** [review]
Maybe some common functions for all distribution scripts as `relayerPost`, `addToMap`, can be extracted

File: scripts/distribute-lp-rewards/cli.js (hunk 1 above)
Comment (samparsky) [reply scripts/distribute-lp-rewards/cli.js:37]: yes definitely, we haven't actually decided on the values yet though

File: test/distribute-lp-reward.js (hunk 3 above)
Comment (samparsky) [reply test/distribute-lp-reward.js:130]:
```
`testBalancerTransferEvents` and `testUniswapTransferEvents` are absolutely the same.
can be declared as one var.
```
yeah the values are the same but the properties are different

**Comment by Ivshti** [comment]
closing cause we're doing a SC-based farm instead: https://github.com/AdExNetwork/adex-staking/issues/73

**Pull request closed without merging**